File: 백준/Silver/7568. 덩치/덩치.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

    int n;
    vector<int> weight, height;
    cin >> n;
    
    vector<int> score(n);

    for (int i = 0; i < n; i++) {
        int x, y;
        cin >> x >> y;
        
        weight.push_back(x);
        height.push_back(y);
    }

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (weight[i] < weight[j] && height[i] < height[j])
                score[i]++;
        }
    }

    for (auto i : score)
        cout << i + 1 << " ";
    
    return 0;
}

File: 백준/Silver/7568. 덩치/덩치.py
import sys

n = int(sys.stdin.readline())
weight = []
height = []
score = [0] * n

for _ in range(n):
    x, y = map(int, sys.stdin.readline().split())
    weight.append(x)
    height.append(y)

for i in range(n):
    for j in range(n):
        if weight[i] < weight[j] and height[i] < height[j]:
            score[i] += 1

for i in score:
    sys.stdout.write(f"{i+1} ")